test/sdn_migration/sdn_migration_test.go

```diff
@@ -0,0 +1,631 @@
+		})
+		Expect(err).ShouldNot(HaveOccurred(), "failed to construct rosa provider")
+		osdProvider, err := osdprovider.New(ctx, ocmToken, clientID, clientSecret, ocmEnv, logger)
+		Expect(err).ShouldNot(HaveOccurred(), "failed to construct osd provider")
+		DeferCleanup(osdProvider.Client.Close)
+
+		if createRosaCluster.MatchesLabelFilter(GinkgoLabelFilter()) {
+			clusterOptions = &rosaprovider.CreateClusterOptions{
+				ClusterName:                  clusterName,
+				Version:                      "4.14.14",
+				UseDefaultAccountRolesPrefix: true,
+				STS:                          true,
+				Mode:                         "auto",
+				ChannelGroup:                 "stable",
+				ComputeMachineType:           "m5.xlarge",
+				Replicas:                     replicas,
+				MultiAZ:                      true,
+				ETCDEncryption:               true,
+				NetworkType:                  "OpenShiftSDN",
+				SkipHealthCheck:              true,
+			}
+		}
+		if os.Getenv("CLUSTER_ID") == "" {
+			if enableClusterProxy.MatchesLabelFilter(GinkgoLabelFilter()) {
+				clusterOptions.HTTPSProxy = os.Getenv("AWS_HTTPS_PROXY")
+				clusterOptions.HTTPProxy = os.Getenv("AWS_HTTP_PROXY")
+				clusterOptions.AdditionalTrustBundleFile = os.Getenv("CA_BUNDLE")
+				clusterOptions.SubnetIDs = os.Getenv("SUBNETS")
+				clusterOptions.NoProxy = "api.stage.openshift.com"
+			}
+			testRosaCluster.id, err = rosaProvider.CreateCluster(ctx, clusterOptions)
+			Expect(err).ShouldNot(HaveOccurred(), "failed to create rosa cluster")
+		} else {
+			testRosaCluster.id = os.Getenv("CLUSTER_ID")
+		}
+
+		rosaCluster, err := osdProvider.ClustersMgmt().V1().Clusters().Cluster(testRosaCluster.id).Get().SendContext(ctx)
+		Expect(err).ShouldNot(HaveOccurred())
+		testRosaCluster.name = rosaCluster.Body().Name()
+		testRosaCluster.version = rosaCluster.Body().Version().RawID()
+		testRosaCluster.channelGroup = rosaCluster.Body().Version().ChannelGroup()
+
+		testRosaCluster.kubeconfigFile, err = rosaProvider.KubeconfigFile(ctx, testRosaCluster.id, os.TempDir())
+		Expect(err).ShouldNot(HaveOccurred())
+
+		testRosaCluster.client, err = openshiftclient.NewFromKubeconfig(testRosaCluster.kubeconfigFile, logger)
+		Expect(err).ShouldNot(HaveOccurred(), "failed to construct service cluster client")
+
+		testRosaCluster.reportDir = fmt.Sprintf("%s/%s", reportDir, testRosaCluster.name)
+		Expect(os.MkdirAll(reportDir, os.ModePerm)).ShouldNot(HaveOccurred(), "failed to create report directory")
+		Expect(os.MkdirAll(testRosaCluster.reportDir, os.ModePerm)).ShouldNot(HaveOccurred(), "failed to create rosa cluster report directory")
+
+	})
+
+	AfterAll(func(ctx context.Context) {
+		if removeRosaCluster.MatchesLabelFilter(GinkgoLabelFilter()) {
+			rosaProvider, err := rosaprovider.New(ctx, ocmToken, clientID, clientSecret, ocmEnv, logger, &aws.AWSCredentials{
+				Region:          region,
+				SecretAccessKey: secretAccessKey,
+				AccessKeyID:     accessKeyID,
+			})
```

Comment:
Shouldn't need to recreate another provider, you can reused the one set up in BeforeAll, right?